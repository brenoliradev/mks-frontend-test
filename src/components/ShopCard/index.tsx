import Image from 'next/image'
import { ProductProps } from 'typings/api'
import {
  Button,
  Description,
  ImageWrapper,
  InfoWrapper,
  NameTag,
  PriceTag,
  Wrapper,
} from './styles'

const ShopCard = ({ description, name, photo, price }: ProductProps) => (
  <Wrapper>
    <ImageWrapper>
      <Image
        width={300}
        height={300}
        src={photo}
        alt={`${name} photo`}
        objectFit="contain"
      />
    </ImageWrapper>
    <InfoWrapper>
      <NameTag>{name}</NameTag>
      <PriceTag>R${price?.toString().split('.00')[0]}</PriceTag>
    </InfoWrapper>
    <Description>{description}</Description>
    <Button>
      {' '}
      <Image
        width={14}
        height={12}
        alt="A Shoping bag Icon"
        src="/img/shopping-bag.svg"
      />
      COMPRAR
    </Button>
  </Wrapper>
)

export default ShopCard
