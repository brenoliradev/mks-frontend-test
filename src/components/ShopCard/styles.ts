import styled from 'styled-components'

export const Wrapper = styled.main`
  width: 100%;
  min-height: 300px;
  max-width: 280px;
  background: #ffffff;
  box-shadow: 0px 2px 8px rgba(0, 0, 0, 0.135216);
  border-radius: 8px;

  display: flex;
  flex-direction: column;
  align-items: center;
  margin: auto;

  padding-top: 18px;

  @media (max-width: 400px) {
    padding-top: 4px;
    min-height: 325px;
  }
`

export const ImageWrapper = styled.section`
  width: 138px;
`

export const InfoWrapper = styled.section`
  width: 100%;
  padding-left: 14px;
  padding-right: 12px;
  min-height: 36px;

  margin: 10px 0 8px;

  display: flex;
  justify-content: space-between;

  @media (max-width: 600px) {
    padding: 0 9px;
    margin: 6px 0 8px;
  }
`

export const NameTag = styled.p`
  font-family: 'Montserrat';
  font-weight: 400;
  font-size: 15px;
  line-height: 19px;
  text-align: left;

  color: #2c2c2c;

  @media (max-width: 600px) {
    font-size: 14px;
    line-height: 16px;
  }
`

export const PriceTag = styled.p`
  background: #373737;
  border-radius: 5px;
  font-family: 'Montserrat';
  font-weight: 700;
  font-size: 15px;
  line-height: 15px;

  display: flex;
  justify-content: center;
  align-items: center;
  padding: 4px 6px;

  color: #ffffff;
`

export const Description = styled.p`
  font-family: 'Montserrat';
  font-weight: 300;
  font-size: 10px;
  line-height: 12px;

  padding: 0 12px;
  margin-top: 2px;
  margin-bottom: 4px;

  color: #2c2c2c;
`

export const Button = styled.button`
  background: #0f52ba;
  border-radius: 0px 0px 8px 8px;
  width: 100%;
  height: 34px;
  inline: none;
  border: none;

  margin-top: auto;

  font-family: 'Montserrat';
  font-weight: 600;
  font-size: 14px;
  line-height: 18px;

  display: flex;
  gap: 14px;
  justify-content: center;
  align-items: center;

  color: #ffffff;

  cursor: pointer;

  background-position: center;
  transition: background 0.7s;

  :hover {
    background: #5785ce radial-gradient(circle, transparent 1%, #0f52ba 1%)
      center/15000%;
  }
  :active {
    background-color: #5785ce;
    background-size: 100%;
    transition: background 0s;
  }
`
