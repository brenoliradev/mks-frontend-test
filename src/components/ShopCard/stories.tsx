import { Story, Meta } from '@storybook/react'

import ShopCard from '.'

export default {
  component: ShopCard,
  title: 'ShopCard',
} as Meta

export const Basic: Story = () => (
  <ShopCard
    brand="a brand testing"
    description="a desc testing"
    id={1}
    name="a name testing"
    photo="https://mks-sistemas.nyc3.digitaloceanspaces.com/products/iphone12x64.webp"
    price={111}
  />
)
