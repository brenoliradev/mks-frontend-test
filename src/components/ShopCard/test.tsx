import { render } from '@testing-library/react'

import ShopCard from '.'

describe('<ShopCard />', () => {
  it('should render the heading', () => {
    render(
      <ShopCard
        brand="a brand testing"
        description="a desc testing"
        id={1}
        name="a name testing"
        photo="https://mks-sistemas.nyc3.digitaloceanspaces.com/products/iphone12x64.webp"
        price={111}
      />,
    )
  })

  it('should render logo correctly', () => {
    const { container } = render(
      <ShopCard
        brand="a brand testing"
        description="a desc testing"
        id={1}
        name="a name testing"
        photo="https://mks-sistemas.nyc3.digitaloceanspaces.com/products/iphone12x64.webp"
        price={111}
      />,
    )

    expect(container.firstChild).toMatchSnapshot()
  })
})
