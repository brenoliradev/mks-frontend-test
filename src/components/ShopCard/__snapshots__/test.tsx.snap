// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ShopCard /> should render logo correctly 1`] = `
.c0 {
  width: 100%;
  min-height: 300px;
  max-width: 280px;
  background: #ffffff;
  box-shadow: 0px 2px 8px rgba(0,0,0,0.135216);
  border-radius: 8px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: auto;
  padding-top: 18px;
}

.c1 {
  width: 138px;
}

.c2 {
  width: 100%;
  padding-left: 14px;
  padding-right: 12px;
  min-height: 36px;
  margin: 10px 0 8px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c3 {
  font-family: 'Montserrat';
  font-weight: 400;
  font-size: 15px;
  line-height: 19px;
  text-align: left;
  color: #2c2c2c;
}

.c4 {
  background: #373737;
  border-radius: 5px;
  font-family: 'Montserrat';
  font-weight: 700;
  font-size: 15px;
  line-height: 15px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 4px 6px;
  color: #ffffff;
}

.c5 {
  font-family: 'Montserrat';
  font-weight: 300;
  font-size: 10px;
  line-height: 12px;
  padding: 0 12px;
  margin-bottom: 4px;
  color: #2c2c2c;
}

.c6 {
  background: #0f52ba;
  border-radius: 0px 0px 8px 8px;
  width: 100%;
  height: 34px;
  inline: none;
  border: none;
  margin-top: auto;
  font-family: 'Montserrat';
  font-weight: 600;
  font-size: 14px;
  line-height: 18px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 14px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #ffffff;
  cursor: pointer;
  background-position: center;
  -webkit-transition: background 0.7s;
  transition: background 0.7s;
}

.c6:hover {
  background: #5785ce radial-gradient(circle,transparent 1%,#0f52ba 1%) center/15000%;
}

.c6:active {
  background-color: #5785ce;
  background-size: 100%;
  -webkit-transition: background 0s;
  transition: background 0s;
}

@media (max-width:400px) {
  .c0 {
    padding-top: 4px;
    min-height: 300px;
  }
}

@media (max-width:600px) {
  .c2 {
    padding: 0 9px;
    margin: 6px 0 8px;
  }
}

@media (max-width:600px) {
  .c3 {
    font-size: 14px;
    line-height: 16px;
  }
}

<main
  class="c0"
>
  <section
    class="c1"
  >
    <span
      style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
    >
      <span
        style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
      >
        <img
          alt=""
          aria-hidden="true"
          src="data:image/svg+xml,%3csvg xmlns=%27http://www.w3.org/2000/svg%27 version=%271.1%27 width=%27300%27 height=%27300%27/%3e"
          style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
        />
      </span>
      <img
        alt="a name testing photo"
        data-nimg="intrinsic"
        decoding="async"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: contain;"
      />
      <noscript />
    </span>
  </section>
  <section
    class="c2"
  >
    <p
      class="c3"
    >
      a name testing
    </p>
    <p
      class="c4"
    >
      R$
      111
    </p>
  </section>
  <p
    class="c5"
  >
    a desc testing
  </p>
  <button
    class="c6"
  >
     
    <span
      style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
    >
      <span
        style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
      >
        <img
          alt=""
          aria-hidden="true"
          src="data:image/svg+xml,%3csvg xmlns=%27http://www.w3.org/2000/svg%27 version=%271.1%27 width=%2714%27 height=%2712%27/%3e"
          style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
        />
      </span>
      <img
        alt="A Shoping bag Icon"
        data-nimg="intrinsic"
        decoding="async"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
      />
      <noscript />
    </span>
    COMPRAR
  </button>
</main>
`;
